File: 1755-defuse-the-bomb/1755-defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int n = code.size();
        vector<int>v(2*n),ans(n,0);
        for(int i = 0;i<2*n;i++){
            int value = code[i%n];
            v[i] = value;
            if(i)v[i] = v[i-1] + v[i];
        }
        int last = n-1, first = n-1-abs(k);

        for(int i=0;i<n;i++){
            if(k>0){
                ans[i] = v[i + k] - v[i];
            }else if(k<0){
                ans[i] = v[last++] - v[first++];
            }else break;
        }
        return ans;
    }
};

File: 1755-defuse-the-bomb/1755-defuse-the-bomb.cs
public class Solution {
    public int[] Decrypt(int[] code, int k) {
        int n = code.Length;
        int[] arr = new int[2*n];
        int[] res = new int[n];
       
        for(int i = 0;i<2*n;i++){
            arr[i] = code[i%n];
            if(i != 0) arr[i] = arr[i-1] + arr[i];
        }
        //for(int i = 0 ;i<2*n;i++)
        //Console.Write(arr[i]+" ");
        int last = n-1,first = n -1 - Math.Abs(k);
        for(int i = 0 ;i <n;i++){
            if(k >0){
                res[i] = arr[i+k] - arr[i];
            }else if(k<0){
                res[i] = arr[last++] - arr[first++];
            }else break;
        }
        return res;

    }
}